overlord/snapstate/handlers.go

```diff
@@ -642,6 +643,28 @@ func hasOtherInstances(st *state.State, instanceName string) (bool, error) {
 	return false, nil
 }
 
+var ErrKernelGadgetUpdateTaskMissing = errors.New("cannot refresh kernel without a gadget update task")
```

Comment:
this is now potentially user visible so it should be a bit more detailed perhaps?

"cannot complete refreshing kernel through an old snapd change without ..." maybe or something similar? 

maybe @anonymouse64 can suggest some even better phrasing

Reply:
fwiw I like your suggestion but happy to adjust if Ian has a better one.